odoo_tools/utils/proj.py

```diff
@@ -25,3 +29,65 @@ def get_current_version(serie_only=False):
     if serie_only:
         ver = ver.split(".")[0]
     return ver
+
+
+def setup_venv(venv_dir, odoo_src_path=None):
+    venv_dir = build_path(venv_dir)
+    if os.path.isfile(os.path.join(venv_dir, "pyenv.cfg")):
+        ui.echo(f"Reusing existing venv {venv_dir}")
+    else:
+        venv.create(venv_dir, with_pip=True)
+    pip = os.path.join(venv_dir, "bin", "pip")
+    if odoo_src_path is None:
+        odoo_src_path = build_path(get_conf_key("odoo_src_rel_path"))
+
+    if not os.path.isfile(os.path.join(venv_dir, "bin", "odoo")):
+        subprocess.call(
```

Comment:
question: any reason not to use https://github.com/camptocamp/odoo-project-tools/blob/master/odoo_tools/utils/os_exec.py#L9 ?

Reply:
no reason, apart from not being aware of this utility function, and it does more than what is needed here. What would be the added value? I don't want to capture output. 

Also I much prefer passing a pre split list of command line arguments to having a string passed to shlex.split, (note that subprocess also accepts a string for args, you just have to pass shell=True and it will call shlex.split for you)

https://docs.python.org/3/library/subprocess.html#frequently-used-arguments